config/routes.rb

```diff
@@ -1,8 +1,12 @@
 Rails.application.routes.draw do
   resources :cars
-  devise_for :users
+  devise_for :users, controllers: { registrations: 'users/registrations' } 
   devise_scope :user do
     get '/users/sign_out' => 'devise/sessions#destroy'
+    get '/users' =>  'devise/registrations#new'
+    get '/users/password' => 'devise/passwords#new'
```

Comment:
I don't understand why we need this (I mean, we can leave it, it's correct, but we don't need it). 
`get '/users/sign_out' => 'devise/sessions#destroy'` works like this by default thanks to `controllers: { registrations: 'users/registrations' }` on line 3. 
Sure, `get '/users'` is new, but I don't understand why we need this route at all, since all our users need authentication to access every action, they will be forced to register, and I don't think we need to explicitly navigate to that route.
`get '/users/password'` would act differently from default, but that would only affect the path in the url and rails helper. So unless you really want to see `/users/password` instead of `/users/password/new`, which is way better for REST convention, we should delete that. But mind that if you do delete it, you'll need to change `user_password_path` helper occurences to `new_user_password_path`

Reply:
The reason for adding that get '/users' =>  'devise/registrations#new' was that when you're in edit profile page of devise if you enter incorrect information it automatically redirects you to http://localhost:3000/users and if you refresh the page it will give you no route matches. I don't know how often that sequence of 2 actions could happen, but it'll result to a error.
I'll delete that line "get '/users/password' => 'devise/passwords#new'" it's really a lot better.

Reply:
Ah, I see. I didn't remember that little detail about user registration with devise

Reply:
The session#destroy expects :delete method, that's why it isn't working.
It can be done like ` link_to "Logout", destroy_user_session_path, data: { turbo_method: :delete } `